src/daedalus_deploy/daedalus_constellation.py

```diff
@@ -58,9 +60,49 @@ def __init__(self, cfg, use_vault):
         web_app_env = {"DATABASE_URL": db_url, "NUXT_R_API_BASE": f"http://daedalus-api:{cfg.api_port}/"}
         web_app = constellation.ConstellationContainer("web-app", cfg.web_app_ref, environment=web_app_env)
 
-        # 6. proxy
+        # 6. acme-buddy
+
+        acme_buddy_staging = os.environ.get("ACME_BUDDY_STAGING", 0)
+        acme_env = {
+            "ACME_BUDDY_STAGING": acme_buddy_staging,
+            "HDB_ACME_USERNAME": cfg.hdb_username,
+            "HDB_ACME_PASSWORD": cfg.hdb_password,
+        }
+        acme_mounts = [
+            constellation.ConstellationVolumeMount("daedalus-tls", "/tls"),
+            constellation.ConstellationBindMount("/var/run/docker.sock", "/var/run/docker.sock"),
+        ]
+
+        acme = constellation.ConstellationContainer(
+            name="acme-buddy",
+            image="ghcr.io/reside-ic/acme-buddy:main",
+            ports=[2112],
+            mounts=acme_mounts,
+            environment=acme_env,
+            args=[
+                "--domain",
+                cfg.proxy_host,
+                "--email",
+                "reside@imperial.ac.uk",
+                "--dns-provider",
+                "hdb",
+                "--certificate-path",
+                "/tls/certificate.pem",
+                "--key-path",
+                "/tls/key.pem",
+                "--account-path",
+                "/tls/account.json",
+                "--reload-container",
+                f"{cfg.container_prefix}-proxy-1",
```

Comment:
Not sure this should have a `-1` suffix. Looking at the constellation source code I don’t think it sets that anywhere.

if you define the proxy object first then you’ll be able to call `proxy.name_external(cfg.container_prefix)` instead of doing a string interpolation here.

Reply:
Good spot - the proxy container here is just called daedalus-proxy, so I had that assumption wrong. Will use the above...!